module TestFailTryReturnDrop {
   err X;

   fun TestFunApply[<Copy, Drop> A](consume a:A):(c:A) {
       c = try {
          b = a;
          discard b;
          return b;
       } catch {
         .X => {
           b = a;
           return b;
         };
       };
       return c;
   };
}